### DISCUSSION TITLE
협업 규칙 같이 정해봐요! 😃

### DISCUSSION THREAD

**Discussion by daadaadaah** [CodeSoom/DDD-Kurly-Clone-Promotion#5] [General]
>일단 제가 생각하기에 이런 규칙이 있었으면 좋겠다는 리스트입니다!ㅎㅎ
혹시, 추가로 이런 규칙이 필요할 것 같은 거 있으면 댓글 달아주세요!!

1. 브랜치 전략
- git flow🚀 vs github flow 🎉  vs 다른 의견 ♥️
2. PR Merge 전략
- @etff 님의 제안(#1) : 찬성 👍  vs 반대 👎

3. PR, Issue 템플릿 도입여부
- 찬성 👍  vs 반대 👎 
- 찬성이라면, 템플릿 어떤걸 추가하는게 좋을지 / 반대라면, PR에 이정도는 이야기줬으면 좋겠다 라는거 같이 이야기해주시면 좋을 것 같아요!

4. 한글 또는 영어 사용 여부
- 전부 통일 : 한글 🚀 vs 영어 🎉
- 상황(테스트 코드, 주석, README, Issue, PR) 마다 다르게 : 이건 댓글로 남겨주세요!ㅎㅎ

5. commit 컨벤션
- 한글 🚀 vs 영어 🎉
- commit 템플릿 사용 하자 ♥️ vs 사용하지 말자 👎
- 일단 예시 템플릿 올렸습니다. 혹시 템플릿 수정이나 다른 의견 있으시면, 의견 남겨주세요!!

6. conflict 발생시, 어떻게 해결할 것인지
- conflict 마주한 사람이 해결 🎉  vs 다른 의견 ♥️

**Comment by daadaadaah** [comment]
## 1. 브랜치 전략
- git flow🚀 vs github flow 🎉 vs 다른 의견 ♥️

**Reply by daadaadaah** [reply]
github flow

**Comment by daadaadaah** [comment]
## 2. PR Merge 전략
- @etff 님의 제안(Git merge rules #1) : 찬성 👍 vs 반대 👎

**Comment by daadaadaah** [comment]
## 3. PR, Issue 템플릿 도입여부
- 찬성 👍 vs 반대 👎
- 찬성이라면, 템플릿 어떤걸 추가하는게 좋을지 / 반대라면, PR에 이정도는 이야기줬으면 좋겠다 라는거 같이 이야기해주시면 좋을 것 같아요!

**Reply by etff** [reply]
남을 설득할 수 있는 최소한 성의면 될 것같아요

**Comment by daadaadaah** [comment]
## 4. 한글 또는 영어 사용 여부
- 전부 통일 : 한글 🚀 vs 영어 🎉
- 상황(테스트 코드, 주석, README, Issue, PR) 마다 다르게 : 이건 댓글로 남겨주세요!ㅎㅎ

**Comment by daadaadaah** [comment]
## 5. commit 컨벤션
- 한글 🚀 vs 영어 🎉
- commit 템플릿 사용 하자 ♥️ vs 사용하지 말자 👎
- 일단 예시 템플릿 올렸습니다. 혹시 템플릿 수정이나 다른 의견 있으시면, 의견 남겨주세요!!- commit 템플릿 
```
# <타입>: <제목(명령문으로, 최대 50글자까지만)>

# <본문>
# - 제목과 본문을 1줄 띄워 분리하기
# - 본문은 1줄에 최대 72글자 내로 제한
# - 본문을 여러줄로 작성할 때 "-"로 구분

# --- COMMIT END ---
# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식 또는 포맷)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# ------------------
```

**Comment by daadaadaah** [comment]
## 6. conflict 발생시, 어떻게 해결할 것인지
- conflict 마주한 사람이 해결 🎉 vs 다른 의견 ♥️

**Reply by daadaadaah** [reply]
- 같이 해결 하기!

**Comment by etff** [comment]
issue나 PR 작성시 레이블을 다는 것이 좋을까요?
필요하다 🎉 vs  하지 않는다 ❤️
